prov/efa/src/rxr/rxr.h

```diff
@@ -825,7 +826,7 @@ int rxr_ep_set_tx_credit_request(struct rxr_ep *rxr_ep,
 				 struct rxr_tx_entry *tx_entry);
 
 void rxr_inline_mr_reg(struct rxr_domain *rxr_domain,
-		       struct rxr_tx_entry *tx_entry);
+		       struct rxr_tx_entry *tx_entry, bool is_shm);
```

Comment:
Can we not tell if this is a shm transfer from the tx_entry or the peer struct? Do we need this signature change?

Reply:
Then inside rxr_inline_mr_reg, we can do  
```
peer = rxr_ep_get_peer(ep, tx_entry->addr);
if (peer->is_local) {}
```
Is this acceptable to you?

Reply:
Yep, that's what I meant.

Reply:
I was thinking to use peer struct too, but it requires the rxr_ep, which is not available here. Even we have mr squash #5664 merged, I think we still need this signature change, as we need to check `if (tx_entry->iov[index].iov_len > rxr_env.max_memcpy_size)` for efa only.

Reply:
We can replace rxr_domain with rxr_ep

Reply:
If replacing rxr_domain with rxr_ep sounds good to you @rajachan, I can update.

Reply:
Registration is generally a domain-level construct, hence the original signature, but you can get to the domain from the EP so that's fine.